### ISSUE TITLE
"incorrect header check" error installing plugins

### ISSUE THREAD

**Issue by jasonzoladz** [LightTable/LightTable#1789]
Here's the error message.  
--------------------------------------------------------------------------------------------
Error: incorrect header check at Zlib._binding.onerror (zlib.js:295:17)
--------------------------------------------------------------------------------------------

(Is this a git issue?)

**Comment by cldwalker** [comment]
Closing since I can't reproduce this without more information

**Comment by mikebelanger** [comment]
Same error when I try and install Gist ([https://github.com/harsha-mudi/lt-gist/issues/new]).  I used fresh install settings by removing everything in ~/Library/Application Support/Lighttable.  I'm on OS X Yosamite 10.10.2, using Lighttable build 0.7.2.

So start with a new install of Lighttable (from command-line) and simply try and install the Gist plugin from the plugin manager.  I get this error:
```
Error: incorrect header check
    at Zlib._binding.onerror (zlib.js:295:17)
```

Interestingly, there's a similar problem installing the haskell plugin too:

https://github.com/jetaggart/light-haskell/issues/52

**Comment by kolya-ay** [comment]
[The same]((https://github.com/otijhuis/paredit-plus/issues/5) with paredit-plus with a new install.

**Comment by mbernat** [comment]
Still happens on 0.7.2.

**Comment by troydm** [comment]
+1 happens when I try to install Zenburn theme plugin on 0.7.2 running on Ubuntu 14.04 64-bit Linux

**Comment by kenny-evitt** [comment]
@jasonzoladz @mikebelanger @kolya-ay @mbernat @troydm can you consistently reproduce this error? Do you get it when installing *any* plugin or is it specific to specific plugins?

**Comment by troydm** [comment]
@kenny-evitt I haven't tried installing any other plugin but I constantly get this error when I try to install this Zenburn theme plugin so it's reproducible

**Comment by kenny-evitt** [comment]
@troydm thanks

**Comment by kolya-ay** [comment]
Confirm I can't still install paredit-plus through plugin manager. However I can install e.g. Subpar plugin. It seems that [the culprit](https://github.com/LightTable/LightTable/blob/master/src/lt/objs/deploy.cljs#L24) is the [obsoleted npm-dependency](https://github.com/LightTable/LightTable/tree/master/deploy/core/node_modules/tar). I use LT form the `atom-shell` branch.

**Comment by kenny-evitt** [comment]
@kolya-ay thanks

What's different about the plugins that can be installed from those that can't that's causing the node tar module to break?

**Comment by MrSpock** [comment]
But still here on OS X El Captain. When installing Go plugin I'm getting;
Error: incorrect header check
    at Zlib._binding.onerror (zlib.js:295:17)

**Comment by kenny-evitt** [comment]
@jasonzoladz @mikebelanger @mbernat @troydm @MrSpock Would you confirm whether you can reproduce this with the *atom-shell* branch?

**Comment by troydm** [comment]
@kenny-evitt just checked with latest version of LightTable from atom-shell branch built from source and the issue is still there :(

**Comment by kenny-evitt** [comment]
@troydm Thanks for checking.

**Comment by rundis** [comment]
Well it seems downloading the tar.gz fails somewhere along the line. For Paredit-plus and go I'm getting a not found basically. Looking at the go plugin it seems to be something whacky with the info in plugin-metadata and actual tags for the project. For paredit-plus I haven't yet been able to figure out why the download fails.

I'll keep on plugin. Seems a little bit more robust error handling would be in place, regardless the cause for this error.

**Comment by rundis** [comment]
**Paredit-plus**
Ok found the problem with paredit-plus. It's metadata lists the source attribute as https://github.com/otijhuis/paredit-plus.git .... the .git part messes everything up when it tries to download https://api.github.com/repos/otijhuis/paredit-plus.git/tarball/0.0.7. 

**Zenburn**
the author has changed his github alias
trying to download: https://api.github.com/repos/wisenomad/zenburn-lighttable-theme/tarball/0.0.1
but if url had been https://api.github.com/repos/zubko/zenburn-lighttable-theme/tarball/0.0.1 it would have worked (although it seems to redirect the url to the one above which doesn't work when calling directly...)

Anyways I'm very quickly approaching a conclusion that this is not really an error in LightTable as such, its stale or incorrect plugin metadata causing us pain and the error-handling sucks a bit to make things worse.

**Comment by cldwalker** [comment]
Fixed and available with 0.8.0-alpha - https://groups.google.com/forum/#!topic/light-table-discussion/rkfHTplo4ss
